cupy/core/core.pyx

```diff
@@ -1233,11 +1233,13 @@ cdef class ndarray:
             array([9998., 9999.])
 
         """
-        if (util.ENABLE_SLICE_COPY and
-            type(slices) is slice and slices == slice(None, None, None) and
-            isinstance(value, numpy.ndarray)):
-            if (self.dtype == value.dtype and
-                    self.shape == value.shape):
+        def does_slice_copy(slices, value):
+            return (type(slices) is slice and
+                    slices == slice(None, None, None) and
+                    isinstance(value, numpy.ndarray))
+
+        if util.ENABLE_SLICE_COPY and does_slice_copy(slices, value):
```

Comment:
- I'm not sure the overhead of the function call is negligible.
- (nit) Break lines before operators.
```suggestion
        if util.ENABLE_SLICE_COPY and (
                type(slices) is slice
                and slices == slice(None, None, None)
                and isinstance(value, numpy.ndarray)
        ):
```

Reply:
Sure!